source/agora/common/TransactionPool.d

```diff
@@ -207,6 +229,19 @@ public class TransactionPool
 
     public void remove (Hash hash) @trusted
     {
+        // delete inputs of transaction from input_map
```

Comment:
You could change the `remove` method to take the `Transaction` as a parameter, and then pass it as an argument from within `updateUTXOSet` in the Ledger. It avoids a lookup in the pool.